src/base/BaseTool.js

```diff
@@ -49,12 +52,55 @@ export default class BaseTool {
     return this._options;
   }
 
+  /**
+   * Merges provided options with existing options
+   *
+   * @memberof BaseTool
+   */
   set options (options) {
-    this._options = options;
+    this._options = Object.assign({}, this._options, options);
   }
 
+  /**
+   * Clears the tools options; preserves internal options, but
+   * with `undefined` values.
+   *
+   * @memberof BaseTool
+   */
   clearOptions () {
     this._options = {};
+    this.internalOptions().forEach((option) => {
+      this._options[option] = undefined;
+    });
+  }
+
+  /**
+   * Internal options that "MUST" be in the tool's options if
+   * certain conditions are met. Method is also good for inspecting
+   * options that can be used to change tool behavior.
+   *
+   * @readonly
+   * @memberof BaseTool
+   */
+  get internalOptions () {
```

Comment:
`get internalOptions()` is perhaps a little confusing. One would expect it to return an options object with current keys/values, whereas you get an array of mandatory option names.